hs_core/templates/pages/genericresource.html

```diff
@@ -87,33 +87,59 @@
         </div>
         {#        ======= Top right buttons =======#}
         <div class="col-sm-12">
-                <div class="lined-wrapper"></div>
-        {% if not metadata_form %}
-            {% if page.perms.change %}
-                <div class="custom-btn-toolbar">
-                    <form class="inline" action="{{ cm.get_absolute_url }}" method="post">
-                        {% csrf_token %}
-                        {% if page.perms.delete %}
-                            <a id="delete" data-toggle="modal" data-target="#delete-resource-dialog">
-                                <span data-toggle="tooltip" data-placement="auto" title="Delete this resource"
-                                      class="glyphicon glyphicon-trash icon-button btn-remove"></span>
-                            </a>
+            <div class="lined-wrapper"></div>
+            {% if not metadata_form %}
+                {% if page.perms.change %}
+                    <div class="custom-btn-toolbar">
+                        {% if not is_owner_user %}
+                            {% if is_mine %}
+                                <form class="pull-right" data-id="form-add-to-my-resources"
+                                      action="/hsapi/_internal/{{ cm.short_id }}/label-resource-action/"
+                                      method="post">
+                                    {% csrf_token %}
+                                    <input type="hidden" name="label_type" value="MINE">
+                                    <input type="hidden" name="action" value="DELETE">
+                                    <button id="btnMyResources" data-toggle="tooltip" data-placement="auto"
+                                            title="Remove from my resources" data-form-id="form-add-to-my-resources"
+                                            class="glyphicon glyphicon-inbox icon-button btn-resource-remove"></button>
+                                </form>
+                            {% else %}
```

Comment:
Concern (an issue for the future): It does not seem that there is actually -- in practice -- any path to this clause. I do not see anything wrong with the clause, but tagging something as "mine" when it is *already* in the my resources list seems somewhat redundant. I would rather limit is_mine tagging to those things are not in the my_resources page already. I do not think this is a real problem and it's fine to leave this in AGU, but I think -- once discovery is working -- that we might eventually want to consider removing it.